src/vm/comsynchronizable.cpp

```diff
@@ -1537,15 +1537,56 @@ void QCALLTYPE ThreadNative::nativeInitCultureAccessors()
     END_QCALL;
 }
 
+typedef HRESULT(WINAPI *pfnSetThreadDescription)(HANDLE hThread, PCWSTR lpThreadDescription);
+extern pfnSetThreadDescription g_pfnSetThreadDescription;
+BOOL g_InitializeSetThreadDescriptionWasRun = FALSE;
+
+
+HRESULT WINAPI InitializeSetThreadDescription(HANDLE hThread, PCWSTR lpThreadDescription)
+{
+    pfnSetThreadDescription func = NULL;
+
+#ifndef FEATURE_PAL
+    if (!g_InitializeSetThreadDescriptionWasRun)
+    {
+        HMODULE hKernel32 = WszLoadLibrary(W("kernel32.dll"));
+        if (hKernel32 != NULL)
+        {
+            func = (pfnSetThreadDescription)GetProcAddress(hKernel32, "SetThreadDescription");
+        }
+        g_InitializeSetThreadDescriptionWasRun = TRUE;
```

Comment:
You may want to use this macro instead - it should save you from the subtle races, etc.: https://github.com/dotnet/coreclr/blob/32f0f9721afb584b4a14d69135bea7ddc129f755/src/inc/delayloadhelpers.h#L96